File: post/src/main/java/ripoff/facebook/post/Post.java
package ripoff.facebook.post;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.File;
import java.nio.file.Path;
import java.time.LocalDateTime;
import java.util.Date;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Builder
public class Post {

    @Id
    @SequenceGenerator(
            name = "post_id_sequence",
            sequenceName = "post_id_sequence"
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "post_id_sequence"
    )
    Long id;
    Long userId;
    String content;
    String attachmentPath; // Jako lista kiedyś.
    LocalDateTime creationDate;
}

File: frontend/src/app/application/service/post.ts
export class Post {
  id!: number | undefined;
  content: string;
  creationDate: string;
  userId?: number;

  constructor(content: string, creationDate: string, userId?: number) {
    this.content = content;
    this.creationDate = creationDate;
    this.userId = userId;
  }
}
